cosi/cosi.md

```diff
@@ -565,15 +589,51 @@ active participant subset Q.
 
 ## Hashing the bitmask in the commitment
 
-To truely bind one signature to a set of signers, the bitmask can be included in
+To truly bind one signature to a set of signers, the bitmask can be included in
 the challenge computation such like H(R || A || bitmask || msg). The signature
 verification process could detect any modifications of the original signature
 before proceeding the computationally expensive process.
 
+## Failure between the challenge and response phase
+
+There exists collective signing protocol variations that would eliminate a DoS
+vulnerability where malicious signers go offline during the challenge/response
+phase. The only way for the protocol to suceed is to restart without that
+malicious signer. If there are F colluding malicious cosigners in the group,
+then they can go offline one-by-one in this way, forcing up to F successive
+restarts, before the honest members of the cosigning group can successfully
+create a valid collective signature. This is of course a limited and potentially
+tolerable DoS attack, “only” O(N) in the group size.  It is therefor important
+to consider whether the scheme should tolerate cosigners failing between the
+"challenge" and "response" phases without requiring restart from scratch. This
+would be another nice-to-have robustness refinement, at the cost of added
+complexity. One possible solution is to make use of Merkle tree for the
+commitments. The leader form all the individual Schnorr commits into a Merkle
+tree, and make the real commit c be the root of that Merkle tree.  In this way,
+the leader is committing not so much to a particular collective Schnorr commit
+but rather to *all possible* Schnorr commits that can be composed from any
+subset of the individual Schnorr commits within that Merkle tree.  See section
+IV.D of the [CoSi](https://arxiv.org/abs/1503.08768) paper for more information.
+
 ## Exception Mechanism
 
 XXX What to do in case a node goes offline, doesn't sign, or doesn't relay up etc. in the tree approach.
```

Comment:
What exactly is an "exception mechanism"?  State the problem: e.g., "Graceful Mechanism to Decline Cosigning", or something like that?